src/briefcase/platforms/linux/appimage.py

```diff
@@ -156,52 +157,69 @@ def build_app(self, app: BaseConfig, **kwargs):
         self.logger.info(f"[{app.app_name}] Building AppImage...")
 
         try:
-            # Build the AppImage.
-            # For some reason, the version has to be passed in as an
-            # environment variable, *not* in the configuration...
-            env = {"VERSION": app.version}
-
-            # Find all the .so files in app and app_packages,
-            # so they can be passed in to linuxdeploy to have their
-            # dependencies added to the AppImage. Looks for any .so file
-            # in the application, and make sure it is marked for deployment.
-            so_folders = set()
-            for so_file in self.appdir_path(app).glob("**/*.so"):
-                so_folders.add(so_file.parent)
-
-            deploy_deps_args = []
-            for folder in sorted(so_folders):
-                deploy_deps_args.extend(["--deploy-deps-only", str(folder)])
-
-            # Build the app image. We use `--appimage-extract-and-run`
-            # because AppImages won't run natively inside Docker.
-            with self.dockerize(app) as docker:
-                docker.run(
-                    [
-                        self.linuxdeploy.appimage_path,
-                        "--appimage-extract-and-run",
-                        f"--appdir={self.appdir_path(app)}",
-                        "-d",
-                        os.fsdecode(
-                            self.appdir_path(app)
-                            / f"{app.bundle}.{app.app_name}.desktop"
-                        ),
-                        "-o",
-                        "appimage",
-                    ]
-                    + deploy_deps_args,
-                    env=env,
-                    check=True,
-                    cwd=self.platform_path,
-                )
-
-            # Make the binary executable.
-            self.os.chmod(self.binary_path(app), 0o755)
+            self._build_appimage(app)
         except subprocess.CalledProcessError as e:
             raise BriefcaseCommandError(
                 f"Error while building app {app.app_name}."
             ) from e
 
+    def _build_appimage(self, app):
+        """Build the AppImage.
+
+        :param app: The application to build
+        """
+
+        # For some reason, the version has to be passed in as an
+        # environment variable, *not* in the configuration...
+        env = {"VERSION": app.version}
+        plugins = []
+        if app.requires:
+            for dependency in app.requires:
+                if "toga-gtk" in dependency or "PyGObject" in dependency:
+                    self.logger.info("Using linuxdeploy GTK plugin")
+                    env["DEPLOY_GTK_VERSION"] = "3"
```

Comment:
Is this envvar something we can reliably set in code? What happens if someone uses GTK4?

Reply:
Right now it is hard coded to support our Toga use case. The plugin does do some automatic detection, but our apps are not successfully being detected. This could be a future PR to extend support for GTK4.

Reply:
Yeah, it's meant to auto-detect but I found it wasn't ... the version would have to be the correct one for the toga-gtk version I suppose (see previous conversation about mapping libs to requirements)

Reply:
The autodetect doesn't work because it uses `ldd` to look for executables that are linked against the GTK libraries. Our only executable is Python itself which is not linked against GTK.